ca_tools/scripts/control_node.py

```diff
@@ -44,142 +39,133 @@ class ctrl_Node:
-            self.stop()
-            self.state=STOP_TURNING
+        if (self._reachedAngle()):
+            self._stop()
+            self._state=STOP_TURNING
             
         else:
             aux = Twist()
-            if (abs(self.my_angleGoal-self.my_pose.theta)>3.14):
+            if (abs(self._my_angleGoal-self._my_pose.theta)>3.14):
                 negateFlag=-1.0
             else:
                 negateFlag=1.0
 
-            if (self.my_angleGoal-self.my_pose.theta)<0:
+            if (self._my_angleGoal-self._my_pose.theta)<0:
                 aux.angular.z=negateFlag*-1.0
             else:    
                 aux.angular.z=1.0
 
             rospy.loginfo("negateFlag is %f",negateFlag)
-            aux.angular.x=0.0
-            aux.angular.y=0.0
-            self.my_pub.publish(aux)
+            self._my_pub.publish(aux)
 
-    def stop_turning(self):
-        self.stop()
-        self.moveForward()
-        self.state=FORWARD
+    def _stop_Turning(self):
+        self._stop()
+        self._moveForward()
+        self._state=FORWARD
 
-    def stop(self):
+    def _stop(self):
             aux = Twist()
-            aux.angular.x=0.0
-            aux.angular.y=0.0
-            aux.angular.z=0.0
-            aux.linear.x=0.0
-            aux.linear.y=0.0
-            aux.linear.z=0.0
-            self.my_pub.publish(aux)
+            self._my_pub.publish(aux)
     
-    def moveForward(self):
+    def _moveForward(self):
         aux = Twist()
-        aux.angular.x=0.0
-        aux.angular.y=0.0
-        aux.angular.z=0.0
         aux.linear.x=0.2
-        aux.linear.y=0.0
-        aux.linear.z=0.0
-        self.my_pub.publish(aux)
-
-    def forward(self):
-        if (self.reachedPosition()):
-            self.stop()
-            self.state=STOP_FORWARD
-            self.goal_num+=1
-            self.goal_num%=4
+        self._my_pub.publish(aux)
+
+    def _forward(self):
+        if (self._reachedPosition()):
+            self._stop()
+            self._state=STOP_FORWARD
+            self._goal_num+=1
+            self._goal_num%=4
         else:
-            self.setGoalAngle()
-            if (not self.reachedAngle()):
-                self.state=STOP_FORWARD
-                self.stop()
+            self._setGoalAngle()
+            if (not self._reachedAngle()):
+                self._state=STOP_FORWARD
+                self._stop()
             else:
-                self.moveForward()
-    
-    def stop_forward(self):
-         self.stop()
-         self.setGoalAngle()
-         self.state=TURNING
-
-    def setGoalAngle(self):
-        self.my_angleGoal=math.atan2((self.my_goals[self.goal_num].y-self.my_pose.y),self.my_goals[self.goal_num].x-self.my_pose.x)
-        rospy.loginfo("Difference between angles is %f,",(self.my_angleGoal-self.my_pose.theta))
+                self._moveForward()
+
+    def _diffAngle(self):
+        return (self._my_angleGoal-self._my_pose.theta)
+
+    def _angleIsBig(self):
```

Comment:
The proper way of doing this is wrapping the angle. If angle > 2pi, angle has to start again from 0. You can do the same logic but wrapping between [-pi, pi].
There is some literature about this.